pkg/validation/test.go

```diff
@@ -808,7 +808,10 @@ func validateCredentials(fieldRoot string, credentials []api.CredentialReference
 		for j, other := range credentials[i+1:] {
 			index := i + j + 1
 			if credential.MountPath == other.MountPath {
-				errs = append(errs, fmt.Errorf("%s.credentials[%d] and credentials[%d] mount to the same location (%s)", fieldRoot, i, index, credential.MountPath))
+				if credential.Name != other.Name {
```

Comment:
```golang
if credential.MountPath == other.MountPath && credential.Name == other.Name {
    errs = append(errs, fmt.Errorf("%s.credentials[%d] and credentials[%d] mount to the same location (%s) and have the same name, which would result in a collision", fieldRoot, i, index, credential.MountPath))
    continue
}
// ...
// validate even more here
```
Does this make sense? Store any error and continue to the next iteration or keep going with the validation.

Reply:
This case would fail:
```
credentials:
  - collection: collection-a
    name: a1
    mount_path: /mnt/secrets
  - collection: collection-a
    name: b1
    mount_path: /mnt/secrets
```
bacause the later code checks if the the `mount_path`s aren't subdirectories of each other, specifically see [this test](https://github.com/openshift/ci-tools/pull/4599/files#diff-2c04428953e5c63ac9f66bfd5fffa4a113f659cd08214c734ab116c05c8c3880R1295-R1301)